app/models/compute_resource.rb

```diff
@@ -52,18 +41,31 @@ class ComputeResource < ActiveRecord::Base
     end
   }
 
-  def self.register_provider(provider)
-    name = provider.name.split('::').last
-    return if supported_providers.values.include?(provider) || supported_providers.keys.include?(name)
-    supported_providers[name] = provider.name
+  def self.supported_providers
+    {
+      'Libvirt'   => 'Foreman::Model::Libvirt',
+      'Ovirt'     => 'Foreman::Model::Ovirt',
+      'EC2'       => 'Foreman::Model::EC2',
+      'Vmware'    => 'Foreman::Model::Vmware',
+      'Openstack' => 'Foreman::Model::Openstack',
+      'Rackspace' => 'Foreman::Model::Rackspace',
+      'GCE'       => 'Foreman::Model::GCE',
+    }
+  end
+
+  def self.registered_providers
+    Foreman::Plugin.all.map(&:compute_resources).inject({}) do |prov_hash,providers|
+      providers.each { |provider| prov_hash.update(provider.name.split('::').last => provider.name) }
+      prov_hash
+    end
   end
 
   def self.providers
-    supported_providers.reject { |p,c| !SETTINGS[p.downcase.to_sym] }.keys
+    (supported_providers.reject { |p,c| !SETTINGS[p.downcase.to_sym] }.keys + registered_providers.keys).uniq
```

Comment:
Do we want to tread the resources from pluging differently that the core ones? I understand I could just uninstall the plugin. It would be just less surprising for me, if the settings allowed disabling providers regardless they are core or not.